app/date-select/controller.js

```diff
@@ -0,0 +1,34 @@
+const FormWizardController = require('../../common/controllers/form-wizard')
+const { dateRegex } = require('../../common/helpers/url')
+
+const DateRegExp = new RegExp('/' + dateRegex)
+
+class DateSelectController extends FormWizardController {
+  get(req, res, next) {
+    if (req.query.referrer) {
+      req.sessionModel.reset()
```

Comment:
I think this should be handled using a redirect first step like mentioned on the config file. Then you can use a `middlewareSetup` middleware to handle setting the query value to the session.